### PR TITLE
Bump minimum Python to 3.10

### PR DESCRIPTION
**Pull request by mvdwetering**
I noticed there were no releases for the 2023.2 betas .
The Gitbhu action was failing and I noticed it was still using Python 3.9. With 2023.2 Python 3.9 support has been dropped from HA.

This pullrequest is an attempt to upgrade to 3.10.

Currently the PR is just a search for 3.9 and replace/extend with 3.10 and 3.11 where it seems relevant. Lets see if it passes CI

### CODE REVIEW

**Comment by MatthewFlamm** [comment]
Thank you!  3.11 is seemingly having problems with aiohttp.  Let's just bump to 3.10 only?

**Comment by mvdwetering** [comment]
ok, will do 3.10 only.

btw should I update teh CHANGELOG.md file?
It seems it has not been updated in a while and not sure if that change would make it 0.13.x

**Comment by MatthewFlamm** [comment]
On my to do list is have releases on GitHub instead of the README.  In the meantime, feel free to add to readme, or I will before releasing. I think this would require a minor version bump as it will no longer be guaranteed to be backwards compatible.

File: .github/workflows/generate_package.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
       - name: setup python
         uses: actions/setup-python@v2
         with:
-          python-version: 3.9
+          python-version: 3.10
```
Comment (MatthewFlamm) [inline .github/workflows/generate_package.yml:16]:
```suggestion
          python-version: '3.10'
```

File: .github/workflows/automatic_generation.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@ jobs:
       - name: setup python
         uses: actions/setup-python@v2
         with:
-          python-version: 3.9
+          python-version: 3.10
```
Comment (MatthewFlamm) [inline .github/workflows/automatic_generation.yml:24]:
```suggestion
          python-version: '3.10'
```

File: .github/workflows/automatic_generation.yml [hunk 3]
```diff
@@ -57,7 +57,7 @@ jobs:
     if: needs.generate_package.outputs.need_to_release == 'true'
     strategy:
       matrix:
-        python-version: [3.9]
+        python-version: [3.10]
```
Comment (MatthewFlamm) [inline .github/workflows/automatic_generation.yml:60]:
```suggestion
        python-version: ['3.10']
```

File: setup.py [hunk 4]
```diff
@@ -30,8 +30,8 @@
         "Intended Audience :: Developers",
         "License :: OSI Approved :: MIT License",
         "Programming Language :: Python",
-        "Programming Language :: Python :: 3.9",
         "Programming Language :: Python :: 3.10",
+        "Programming Language :: Python :: 3.11",
```
Comment (MatthewFlamm) [inline setup.py:34]:
```suggestion
```

**Comment by mvdwetering** [comment]
Manually running the generate command below made changes for HA 2023.2.0b4 without errors, so I guess it works :) 

```bash
export PYTHONPATH=$PYTHONPATH:$(pwd)
python generate_phacc/generate_phacc.py
```

**Pull request merged**